File: leetcode_py/p1072.py
from typing import Counter, List


class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        cnt = Counter()
        for row in matrix:
            if row[0]:  # 翻转第一个为 1 的 行, 保证所有都为0
                for j in range(len(row)):
                    row[j] ^= 1
            cnt[tuple(row)] += 1
        return max(cnt.values())

File: leetcode_C++/p1072.cpp
/*
@Author: mrxhar
@Date:   2023-09-07 11:02:01
@Email:  1337904145@qq.com
@Blog:
*/
#include <vector>
using std::vector;
#include <string>
using std::string;
#include <unordered_map>
using std::unordered_map;
#include <cmath>
using std::max;

class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
        // 记录的时候 使用 string类型记录还是使用别的
        unordered_map<vector<bool>, int> cnt;
        int n = matrix[0].size();
        int ans = 0;
        for (auto& row : matrix) {
            vector<bool> r(n);  // 记录bit位还不够
            for (int j = 0; j < n; j++) {
                r[j] = ((row[j] ^ row[0]));
            }
            ans = max(ans, ++cnt[r]);
        }
        return ans;
    }
};
int main() {
    return 0;
}
